_posts/2024-03-29-geospatial-techno.md

```diff
@@ -0,0 +1,78 @@
+
+## Introduction
+Geospatial data has no intrinsic visual component and it must be styled to be visually represented on a map. GeoServer uses a markup language called Styled Layer Descriptor (SLD) to define styling rules for displaying data. SLD is an XML-based language that allows users and software to control the visual portrayal of geospatial data. This language ensures that clients and servers can both understand how to render the data visually.
+
+**Note.** In this blog post, we used GeoServer version 2.20.0.
+
+## Add a Style
+To add a new style, navigate to the **Data > Styles** page, then click on the **Add a new style** link. You will be redirected to the new style page, which is the same as the Style Editor Data tab.
+
+This tab includes basic style information, the ability to generate a style and legend details. It has some mandatory basic style information, such as:
+- Name: It's the name of the style and it must be a unique name.
+- Workspace: Styles can be inside workspaces which causes restrictions. In other words, the styles in a workspace can only be assigned to the services of that, and other services outside it, cannot use these styles. Styles also can be "global" or no workspace, so they don't have any restrictions and services can be used for all suitable styles.
+- Format: Default options are **SLD** and **ZIP** formats. To use other formats such as **CSS** and **YSLD**, you should download and install extensions. Make sure to match the version of the extension to the version of the GeoServer.
+
+The "Style Content" area provides options for creating, copying, or uploading a style. It has three options:
+- Generate a default style: Choose a generic style based on geometry such as Point, Line, Polygon, Raster, or Generic and click the **Generate** link when selected.
+- Copy from existing style: Select an existing style from GeoServer and copy its contents to the current style. Note that not all styles may be compatible with all layers. Click the **Copy** link when selected.
+- Upload a style file: Press the **Browse** button to locate and select a plain text file from your local system to add as the style. Click on the **Upload** link to add the style file.
+
+The **Legend** area allows you to preview the legend for the style. Click on the **Preview legend** link to generate a legend based on the current settings.
+
+At the bottom of the Style Editor page, you'll find several options: **Validate**, **Apply**, **Save** and **Cancel**. During editing and especially after editing is complete, you can check the validation of the syntax by pressing the **Validate** button at the bottom. If any validation errors are found, they will be displayed a red message and if no errors are found, you will see the green message. To make changes, press the **Apply** button to access all the tabs and finally press the **Save** button.
+
+After having created the style, it's time to apply it to the layer. To do it, follow these steps:
```

Comment:
Alternatively, switch to the Publishing tab within Styles and rapidly assign it to multiple layers there

Reply:
Yes, this method is also correct. I explained this method in this blog post and I will discuss the alternative method in my upcoming post.